src/wp-admin/includes/image.php

```diff
@@ -278,70 +282,98 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 */
 		$threshold = (int) apply_filters( 'big_image_size_threshold', 2560, $imagesize, $file, $attachment_id );
 
-		// If the original image's dimensions are over the threshold,
-		// scale the image and use it as the "full" size.
-		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+		// Generate full sized images for all mime types.
+		foreach ( $output_mime_types as $mime_index => $output_mime_type ) {
 			$editor = wp_get_image_editor( $file );
 
 			if ( is_wp_error( $editor ) ) {
 				// This image cannot be edited.
 				return $image_meta;
 			}
 
-			// Resize the image.
-			$resized = $editor->resize( $threshold, $threshold );
-			$rotated = null;
-
-			// If there is EXIF data, rotate according to EXIF Orientation.
-			if ( ! is_wp_error( $resized ) && is_array( $exif_meta ) ) {
-				$resized = $editor->maybe_exif_rotate();
-				$rotated = $resized;
-			}
```

Comment:
we do need to re-initialize the editor to get back to the original image, for example after creating the cropped or rotated version. otherwise the editor will perform subsequent edits with that image.

Reply:
@adamsilverstein How do you get to that conclusion? [`_wp_make_subsizes()`](https://developer.wordpress.org/reference/functions/_wp_make_subsizes/) reuses the same editor instance to generate all sub-sizes, so I don't see why we wouldn't be able to do the same here?

Reply:
👍🏼